_application_development/workflow.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Workflow
+position: 3
+---
+
+# The Development Workflow
+
+The end goal is for the commit history in `master` to tell a logical and descriptive story about the code
+
+Before moving on, find 30 minutes to [watch Stephen Ball's _Deliberate Git_ talk](https://vimeo.com/72762735). This will help you understand how we can make our web development more efficient and sustainable by following this workflow.
+
+This guide will cover 3 facets of the workflow:
+
+1. Writing code
+2. Generating a clean and thorough commit history
+3. Reviewing and merging code into deployable branches via Pull Requests
+
+## The coding process
+
+### Always do your work in a feature branch
+
+Remember the 1 rule to, uh, _rule_ them all? Yeah, anything on `master` is deployable. So your work-in-progress should not be done on `master`. That's where feature branching comes in.
+
+1. Start with a clean `master` branch that's deployed on production.
+
+   ```
+   $ git checkout master
+   $ git pull origin master
+   ```
+
+2. Create your feature branch using a semantically-descriptive name with hyphens as separators
+   Good examples: `ci-integration` or `homepage-refactor`
+   Bad examples: `matt` or  `updates`
+
+   ```
+   $ git checkout -b my-feature
+   Switched to a new branch 'my-feature'
+   ```
+
+### Make incremental commits throughout your process
+
+As part of your flow it's important to create separate commits for logical groups of code. For example, let's say you're doing some frontend work on a homepage, and as part of that effort you decide to replace a static styles.css stylesheet with a SASS pipeline. In that case you have at least two related but logically distinct sets of changes: The SASS structure and associated dependences, and then the UI changes that have to be made in the CSS and HTML. Once you had built the SASS pipeline you would want to create a commit for that change, probably titled something like "Implement SASS". Next you would move on to your UI updates and roll those into a subsequent commit (which likely touches some of the same files), perhaps called "Update homepage widget styling".
```

Comment:
with big changes like this example I like to make another branch from my branch and merge it back in